src/sentry/static/sentry/app/views/performance/vitalDetail/vitalPercents.tsx

```diff
@@ -1,44 +1,67 @@
 import React from 'react';
 import styled from '@emotion/styled';
 
+import Tooltip from 'app/components/tooltip';
 import {IconCheckmark, IconFire, IconWarning} from 'app/icons';
-import {t} from 'app/locale';
+import {t, tct} from 'app/locale';
 import space from 'app/styles/space';
+import {WebVital} from 'app/utils/discover/fields';
 import {formatPercentage} from 'app/utils/formatters';
 import theme, {Color} from 'app/utils/theme';
 
-import {VitalState, vitalStateColors} from './utils';
+import {VitalState, vitalStateColors, webVitalMeh, webVitalPoor} from './utils';
 
 type Percent = {
   vitalState: VitalState;
   percent: number;
 };
 
 type Props = {
+  vital: WebVital | WebVital[];
   percents: Percent[];
   showVitalPercentNames?: boolean;
 };
 
+const vitalStateIcons = {
+  [VitalState.POOR]: IconFire,
+  [VitalState.MEH]: IconWarning,
+  [VitalState.GOOD]: IconCheckmark,
+};
+
+function vitalStateText(vital, vitalState) {
+  switch (vitalState) {
+    case VitalState.POOR:
+      return Array.isArray(vital)
+        ? t('Poor')
+        : tct('Poor: >[threshold]ms', webVitalPoor[vital]);
+    case VitalState.MEH:
+      return Array.isArray(vital)
+        ? t('Needs improvement')
+        : tct('Needs improvement: >[threshold]ms', webVitalMeh[vital]);
+    case VitalState.GOOD:
+      return Array.isArray(vital)
+        ? t('Good')
+        : tct('Good: <[threshold]ms', webVitalMeh[vital]);
+    default:
+      return null;
+  }
+}
+
 export default function VitalPercents(props: Props) {
   return (
     <VitalSet>
       {props.percents.map(p => {
+        const VitalStateIcon = vitalStateIcons[p.vitalState];
```

Comment:
`p` is a confusing var name in this case - we use `p` a lot in styled components to prefer to `props`. `percent` would make more sense here (or even destructuring it so that you don't need to deal with an intermediate name).